src/main/java/org/csanchez/jenkins/plugins/kubernetes/pod/retention/Reaper.java

```diff
@@ -130,29 +142,25 @@ private void activate() {
 
     @Override
     public void eventReceived(Watcher.Action action, Pod pod) {
-        if (action == Watcher.Action.DELETED) {
-            String ns = pod.getMetadata().getNamespace();
-            String name = pod.getMetadata().getName();
-            Jenkins jenkins = Jenkins.getInstanceOrNull();
-            if (jenkins != null) {
-                for (Node n : new ArrayList<>(jenkins.getNodes())) {
-                    if (!(n instanceof KubernetesSlave)) {
-                        continue;
-                    }
-                    KubernetesSlave ks = (KubernetesSlave) n;
-                    if (Objects.equals(ks.getNamespace(), ns) && Objects.equals(ks.getPodName(), name)) {
-                        LOGGER.info(() -> ns + "/" + name + " was just deleted, so removing corresponding Jenkins agent");
-                        try {
-                            jenkins.removeNode(ks);
-                            return;
-                        } catch (Exception x) {
-                            LOGGER.log(Level.WARNING, "failed to reap " + ns + "/" + name, x);
-                        }
-                    }
-                }
-            }
-            LOGGER.fine(() -> "received deletion notice for " + ns + "/" + name + " which does not seem to correspond to any Jenkins agent");
```

Comment:
Moved to https://github.com/jenkinsci/kubernetes-plugin/pull/772/files#diff-55ff61b56516ad2fa5dd36df80c3836cR195